game-core/src/main/java/tools/map/making/MapProperties.java

```diff
@@ -78,10 +78,7 @@ public void writePropertiesToObject(final List<MapPropertyWrapper<?>> properties
     return colorMap;
   }
 
-  public void setColorMap(final Map<String, Color> value) {
-    colorMap = value;
-  }
-
+  @SuppressWarnings("unused")
```

Comment:
You should probably add a comment somewhere to indicate this is invoked via reflection

Reply:
I agree, this leads to a problem I've been thinking about of how to document many suppression annotations. It's more an issue when the annotations are in many files compared to just one. 

In this case I would think that there should be a comment on how the reflection works and that it is there. I'm not sure if there is a good way to tie each suppression to the class comment to indicate that, or if it's simply better for each and every suppression to write it as:
```
@SuppressWarnigns("unused") // reflective-access
```

I kinda like that in some ways, the comment seems to be a hack though. Perhaps we should consider creating annotations to track these kinds of 'meta-comments'? For example:
```
@AccessedByReflection
@SuppressWarnigns("unused")
```

Any thoughts?

Reply:
(updates to this file were reverted:  https://github.com/triplea-game/triplea/pull/4609/commits/9349c6f3fadee33ffc5182622403f99adf7e1b27)